src/rest.js

```diff
@@ -66,7 +66,7 @@ function sql(datasetKey, sqlQuery, lastRefresh, sampleExtraction) {
     }
 }
 
-var datasetRegex = /^(https?:\/\/data\.world\/)?(.+\/.+)$/;
+var datasetRegex = /^(https?:\/\/[a-z0-9-.]*data\.world\/)?(.+\/.+)$/;
```

Comment:
I think it could be a good idea to add a unit test, and then supply a range of inputs. For example, as-is, this regex will match:

- `https://...data.world/`
- `https://-data.world/`
- `https://mybaddomaindata.world/`

I'm actually not sure I even like the fact that SSL is optional. I don't want to break the existing integration, but I feel like it should not break the existing integration if we make `https` required instead of also allowing `http`.

I don't think there's much potential harm in matching more than needed, since we always go to `api.data.world` regardless of the input domain. But, it does feel very broad.